resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -966,7 +971,12 @@ private void finishResolve(Promise<List<T>> promise, Throwable cause) {
         if (finalResult != null) {
             if (!promise.isDone()) {
                 // Found at least one resolved record.
-                DnsNameResolver.trySuccess(promise, filterResults(finalResult));
+                final List<T> result = filterResults(finalResult);
+                if (!DnsNameResolver.trySuccess(promise, result)) {
+                    for (T item : result) {
+                        ReferenceCountUtil.safeRelease(item);
```

Comment:
Should I remove the content of `finalResult`? Also, I'm not sure if `finalResult` will be reused or not.

Reply:
its fine to just keep it...